apps/web/vibes/soul/primitives/dynamic-form/schema.ts

```diff
@@ -5,6 +5,7 @@ interface FormField {
   label?: string;
   errors?: string[];
   required?: boolean;
+  id?: string;
```

Comment:
Is it necessary to add this here vs it being handled by the generics?

Reply:
I'm not sure what you mean here?

Reply:
We've been using generics in the props so that data can be passed to the component and back to the server via server actions and I was thinking that this could be an extra field like that. Perhaps I'm misunderstanding, though, and this `id` field is used by VIBES? It being optional jumped out at me as something that might not be needed

Reply:
So this was more for the consumers needing it. In our consumption, we were conflating `name` with the entity ID, and this doesn't work well when trying to build payloads for the API using `fields` on the state (specifically BigCommerce Custom Form Fields). This separates trying to identify form fields from the `name` to the `id` here.

Reply:
I see. I think that we might be able to accomplish that by allowing extra fields if the fields are generic. I don't want to continue dragging this along, though, so again, this is something we can refine later

Reply:
`placeholder` is another attribute example of something that would be nice to add onto fields here.

Reply:
Agreed!